scripts/ci-cdk-deploy.sh

```diff
@@ -0,0 +1,43 @@
+#!/bin/bash
+
+set -e
+
+cdk_app_dir="cdk-app" # it is expected that the CDK app in a repository is found from this folder
```

Comment:
Can we merge the logic of the this script as an additional branch inside `for module in "${MODULES[@]}"` of https://github.com/ringier-data/ops-ci-shared/blob/aa1dab28c8a7be8b1c7e081e593cc79e5bdc2011/scripts/ci-deploy.sh#L38? I don't see a big value to make this a separate file, because 40% of a 50-lines-script is redundant with that one.

My proposal:
```
for module in modules {
   if package.json and package-lock.json exist {
      run the three lines for CDK 
   } else {
      run the existing Ansible logic
  }
}
```

The cleaner code could also generate confusion, because the env var specifying the IaC location is called `ANSIBLE_FOLDER`, and we cannot simply rename that one because of backward compatibility. My proposal:
```
   // insert after L23
   if INFRASTRUCTURE_FOLDERS does not exist {
      INFRASTRUCTURE_FOLDERS = ANSIBLE_FOLDERS
   }
```
In addition:
*  `MODULES=("${ANSIBLE_FOLDERS//,/ }")` will become `MODULES=("${INFRASTRUCTURE_FOLDERS//,/ }")`
*  Ansible-specific things like `pip` and `ansible-galaxy` will go into the Ansible logic branch mentioned above. It is not perfect due to the edge case that ANSIBLE_FOLDERS actually specifies multiple Ansible deployments, the `pip` and `ansible-galaxy` run will be executed multiple times. This hurts nothing, but costs several seconds more.

Reply:
I think this is a good idea. My initial approach was to first have a separate deployment script to not mess with anything existing and then merge it a bit later on. But might as well do it now :)

I implemented your proposal. One thing to note with the approach I did is that if a CDK app folder is detected then that overrides any Ansible related deployments. The deployment could of course be made even more flexible by allowing multiple CDK apps and multiple Ansible modules and deploying whatever is found but preferably I would implement it when a use case arises since there's more effort in testing it.